spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2636,7 +2636,11 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
   def nullIfWhenPrimitive(expression: Expression): Expression = if (isPrimitive(expression)) {
     val zero = Literal.default(expression.dataType)
-    If(EqualTo(expression, zero), Literal.create(null, expression.dataType), expression)
+    expression match {
+      case _: Literal if expression != zero => expression
```

Comment:
Is this related?

Reply:
Indirectly related, this will avoid to put `EqualTo` for non zero literals for divisions